File: magma-platform/magma-platform-base/src/main/java/de/jalumu/magma/platform/base/application/MagmaApplicationBase.java
package de.jalumu.magma.platform.base.application;

import net.kyori.adventure.platform.AudienceProvider;
import revxrsal.commands.CommandHandler;

public interface MagmaApplicationBase {

    void initialize();
    void start();
    void shutdown();
    CommandHandler getCommandHandler();
    AudienceProvider getAudience();

}

File: magma-platform/magma-platform-base/src/main/java/de/jalumu/magma/platform/base/application/MagmaApplicationBase.kt
package de.jalumu.magma.platform.base.application

import net.kyori.adventure.platform.AudienceProvider
import revxrsal.commands.CommandHandler

interface MagmaApplicationBase {
    fun initialize()
    fun start()
    fun shutdown()
    val commandHandler: CommandHandler?
    val audience: AudienceProvider?
}
